frontend/src/app/main/ui/modal.cljs

```diff
@@ -1,61 +1,126 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; This Source Code Form is "Incompatible With Secondary Licenses", as
+;; defined by the Mozilla Public License, v. 2.0.
+;;
+;; Copyright (c) 2020 UXBOX Labs SL
+
 (ns app.main.ui.modal
   (:require
-   [cuerdas.core :as str]
+   [okulary.core :as l]
    [goog.events :as events]
    [rumext.alpha :as mf]
    [app.main.store :as st]
    [app.main.ui.keyboard :as k]
-   [app.util.data :refer [classnames]]
-   [app.util.dom :as dom])
+   [app.util.dom :as dom]
+   [app.main.refs :as refs]
+   [potok.core :as ptk])
   (:import goog.events.EventType))
 
-(defonce state (atom nil))
+(defonce components (atom {}))
+
+(defn show-modal [id type props]
+  (ptk/reify ::show-modal
+    ptk/UpdateEvent
+    (update [_ state]
+      (-> state
+          (assoc-in [:workspace-local :modal] {:id id
```

Comment:
I think the modal should be on the root, becuase we can have modals outside of workspace, and workspace local is a local state of the page...

Reply:
OK